### DISCUSSION TITLE
How to bind a physical key to a specific button elem？

### DISCUSSION THREAD

**Discussion by goddade** [ImpulseAdventure/GUIslice#552] [Q&A] [answered]
or how to press a button in a program?

**Comment by Pconti31** [comment]
@goddade It's easy enough to do but there is no documentation.   The only example is [ex04_ard_ctrls_pin](https://github.com/ImpulseAdventure/GUIslice/tree/master/examples/arduino/ex04_ard_ctrls_pin)

Paul--

**Reply by goddade** [reply]
In this example, the physical key is used as a global input method, but I want to bind a physical key to a button.
In other words, is there an event of pressing a button?

**Comment by Pconti31** [comment]
@goddade Hmmm, that does require a bit of a think...
Can you give an example of what you want from your UI?
Will there be more than one page or just the main page?
Paul--

**Reply by goddade** [reply]
There is a physical key and a button. When the physical key is pressed, the button also has a pressing effect.

**Comment by Pconti31** [comment]
@goddade 
The answer to your querstion is that there is no direct api to help you.
However, there are low level calls that might be used to create support.
Creation of such support would require more information about your planned GUI.

1 - Will your physical button be tied to only one on screen button?

2 - Can a user still trigger the on screen button without using the physical button?

3 - Do you plan on having more than one physical button? 
    Each tied to one specific on screen button?
    If so, how many?
    
4 - Will your GUI have more than one page?

5 - Will your on screen button be on a Base Page?

6 - Last why do you want a user to have the option of using a physical button?

The answer to these questions make a possible solution easy or hard or impossible...

Paul--

**Reply by goddade** [reply]
1 - Will your physical button be tied to only one on screen button?
Yes.

2 - Can a user still trigger the on screen button without using the physical button?
No touch screen or other input method.

3 - Do you plan on having more than one physical button?
Each tied to one specific on screen button?
If so, how many?
6 touch keys with 6 buttons.

4 - Will your GUI have more than one page?
Only one page.

5 - Will your on screen button be on a Base Page?
Yes.

6 - Last why do you want a user to have the option of using a physical button?
No touch screen or other input method.

My device has a screen with 3 touch keys on each side. Next to each touch key, there is a button on the screen.
If the touch key is pressed, the button will have a pressed effect as a prompt.

**Comment by Pconti31** [comment]
@goddade I worked out a simple solution using EasyButton that should do what you want.
I'll post it on my [GUIslice-Solutions](https://github.com/Pconti31/GUIslice-Solutions) Repository when completed.
I'll give an update here when its ready.
Paul--

**Comment by Pconti31** [comment, marked as answer]
@goddade The sample is finished.
checkout [btn-physical](https://github.com/Pconti31/GUIslice-Solutions/tree/main/apps/btn-physical)
If it meets your needs please mark this discussion as answered.
Paul--

**Reply by goddade** [reply]
Thank you!
